lumen/ai/agents.py

```diff
@@ -332,12 +332,16 @@ def format_float(num):
             else:
                 return f"{num:.1e}"  # Exponential notation with two decimals
 
-        if len(df) < 100:
+        length = len(df)
+        if length < 100:
```

Comment:
Should df.size be used here? I believe len() only checks rows while size checks cols and rows.

Reply:
Feel free to fix.